pkg/transport/pool.go

```diff
@@ -0,0 +1,95 @@
+package transport
+
+import (
+	"errors"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+const paramTransportClientTimeout = "client-timeout"
+const paramTransportType = "type"
+
+const transportTypeHttp = "http"
+
+const defaultTransportClientTimeout = 10 * time.Second
+const defaultTransportType = transportTypeHttp
+
+// TransportPool creates http.Clients as required, using the provided viper.Viper for configuration.
+type TransportPool struct {
+	config *viper.Viper
+	logger logrus.FieldLogger
+
+	mu      sync.Mutex
+	clients map[string]*Client
```

Comment:
You can replace the mutex and map with `sync.Map` (available since 1.9) which is optimised for this write-once-read-many case

Reply:
Generally speaking, I'm not a fan of the `sync.Map`, just because it's all weakly typed.  When it's not critical (such as this), I prefer simple.

In practice it's likely to only ever be 1 read + 1 write, and the lock only exists so I don't need to check if there's anything that's not safe.